### ISSUE TITLE
Check if family has siblings

### ISSUE THREAD

**Issue by m4rc1e** [googlefonts/fontbakery#1568]
Some families also have SC siblings, https://github.com/google/fonts/pull/1011#issuecomment-326927195.

**Comment by davelab6** [comment]
Could you describe more what is needed to happen within this check?

**Comment by felipesanches** [comment]
Let's try to be more verbose in the descriptions of proposed new tests.

For instance I'd suggest we try to agree on the long description of a test that will be placed in its doc-string.
This should be a short sentence followed by a more detailed description of the kind of problem it tries to detect. If the problem is not hypothetical, then it would be desirable to have an actual example of a bad font or family (and maybe also of a good one). It could be, for example, a URL to a google/fonts issue.

**Comment by felipesanches** [comment]
My understanding here is that Marc wants to get alerts when a SC sibling exists because it would be desirable to update both families at once, to reduce/avoid redundant work.

But I still don't see how that concern could be effectively made into an actual fontbakery check.

**Comment by m4rc1e** [comment]
> Could you describe more what is needed to happen within this check?
Updated the initial issue

**Comment by m4rc1e** [comment]
@felipesanches we submitted at exactly the same time :-)

**Comment by felipesanches** [comment]
Is the proposal here to make sure that if an SC or One sibling is found on GF API then it should have the same version ?

Or would you phrase it differently? What's the testing strategy proposed here ?

**Comment by graphicore** [comment]
I wonder how this check fails or passes.

**Comment by felipesanches** [comment]
@m4rc1e This seems to potentially be yet another fontbakery test on the proposed "superfamilies" category, like @davelab6 mentioned at issue #1487

**Comment by felipesanches** [comment]
@m4rc1e Inspired by this new-test request and also by our recent conversation during ATypI regarding having proper descriptions and rationales for tests in FontBakery, I'd like to ask for more information to be provided at this page:

https://github.com/googlefonts/fontbakery/wiki/com.google.fonts-test-166

I'd like to establish this as a better process for requesting new Font Bakery tests. There will be similar pages for all other tests at:
https://github.com/googlefonts/fontbakery/wiki/Documentation-for-the-Google-Fonts-Specification

**Comment by felipesanches** [comment]
I agree with Dave that setting up the wiki was a bad idea. So let's forget about the wiki and simply try to provide detailed rationale descriptions on each new request for new checks.

**Comment by felipesanches** [comment]
I'm removing the wiki pages to avoid confusion.

**Comment by felipesanches** [comment]
Let's imagine a usage scenario on the command line:

* The user invokes `check-googlefonts` on `~/gfonts/ofl/vollkorn/*.ttf`
* FB-Core will end up validating that the version of the local copy of the Vollkorn family under testing is greater than what's on GFonts production servers as well as what's on the github repo (tha'ts what **check/117** does (see #1640).
* In order to do so, it does grab a copy of the files on the google/fonts repo on github, And also from production servers. See: https://github.com/googlefonts/fontbakery/blob/17d947d8dc967fe1ceb0a0a0f07f8405c99aac38/Lib/fontbakery/specifications/googlefonts.py#L4239-L4258
* It may try to fetch a similar URL for the SC variant from github and for prod servers and infer that a variant is already publicly available if the pages do not result in a 404 error.
* Once determined that a prior release of a Vollkorn SC variant indeed exists, FB-Core has to figure out where the local files for those would be placed in the users computer.
* Here we might simply emit a WARN suggesting that the used should invoke `check-googlefont`s passing the whole set of files from both the base family as well as its variants. This unfortunaely could conflict with **check/002**:
https://github.com/googlefonts/fontbakery/blob/17d947d8dc967fe1ceb0a0a0f07f8405c99aac38/Lib/fontbakery/specifications/googlefonts.py#L114-L128
* The other option would be to have FB-Core guess the location of the files for the family variant by modifying the path of the original family files and appending "sc" to the directory name. This euristics may fail unless the whole collection uses the same naming convention for family variant directories.

There is also some trouble with the integration of this with Fontbakery Dashboard. Potentially all of this logic for inferring groups of correlated families risks being fully re-implemented by the frontend code. And it also affects the strategies used for splitting the collection data into separate containers. I guess that base families and their variants would have to be packed together into a single container.

**Comment by graphicore** [comment]
> There is also some trouble with the integration of this with Fontbakery Dashboard.

It's not just the frontend. The dashboard has not really a "location" for collection families (collection can be upstream families, PRs to google/fonts, google/fonts master, the gf-API etc.) Using `google/fonts` or the `gf-API` would just produce false positives in many cases.

I think all this "figure how to get corresponding files from somewhere" is a bad approach. It would be better if we could somehow explicitly define dependencies (e.g. in METADATA.pb). Not sure about details though. This applies to all the upcoming "superfamilies" business that we have.

**Comment by davelab6** [comment]
`com_google_fonts_check_002()` probably needs to become smarter at validating that the set of fonts form a contiguous superfamily; currently it checking that the set of fonts is only forming a contiguous web font family (1-18 styles)

**Comment by davelab6** [comment]
Perhaps whitelisting in the form of fontbakery.yml metadata to link the parent family (eg vollkorn) to its siblings (eg vollkornsc) could make this easier

**Comment by felipesanches** [comment]
I know it is labeled "P0 - Urgent" but this (and any other issue related to super-families) will be doable by next Friday, so I'm moving this to the next minor release milestone.

**Comment by m4rc1e** [comment]
Thanks @felipesanches

**Comment by felipesanches** [comment]
@davelab6 said at #2340 **"Add check for Small Cap siblings"**: 

> We should have a Small Cap (SC) generator checker that goes over all the fonts in the repo, finds ones with SMCP OT code, and generates SC sibling families, and checks if the SC fonts are the same as the generated ones, and if not, warns.
> 
> This is maybe not needed in the regular everyday check profile, but is something that is needed less frequently until it passes, and then needs to become a fail to stop regressions.
> 
> We should also have a check that all SC sibling families have a parent with SC included as SMCP glyphs + code...

I will now close that other issue as a duplicate of this one.

**Comment by felipesanches** [comment]
**quick note:** we do have a merely informative check that lists all siblings of a given family. It's been implemented a few weeks ago:

**com.google.fonts/check/superfamily/list**
